core/src/main/java/org/kiwix/kiwixmobile/core/main/MainPresenter.kt

```diff
@@ -18,32 +18,20 @@
 package org.kiwix.kiwixmobile.core.main
 
 import android.util.Log
+import org.kiwix.kiwixmobile.core.base.BaseContract
 import org.kiwix.kiwixmobile.core.base.BasePresenter
-import org.kiwix.kiwixmobile.core.page.bookmark.adapter.BookmarkItem
 import org.kiwix.kiwixmobile.core.data.DataSource
 import org.kiwix.kiwixmobile.core.di.ActivityScope
-import org.kiwix.kiwixmobile.core.page.history.adapter.HistoryListItem.HistoryItem
 import org.kiwix.kiwixmobile.core.main.MainContract.Presenter
-import org.kiwix.kiwixmobile.core.main.MainContract.View
-import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem.BookOnDisk
+import org.kiwix.kiwixmobile.core.page.bookmark.adapter.BookmarkItem
+import org.kiwix.kiwixmobile.core.page.history.adapter.HistoryListItem.HistoryItem
 import javax.inject.Inject
 
 private const val TAG = "MainPresenter"
 
 @ActivityScope
 internal class MainPresenter @Inject constructor(private val dataSource: DataSource) :
-  BasePresenter<View?>(), Presenter {
-  override fun loadBooks() {
-    compositeDisposable.add(
-      dataSource.languageCategorizedBooks.subscribe(
-        { view?.addBooks(it) }
-      ) { e -> Log.e(TAG, "Unable to load books", e) })
-  }
-
-  override fun saveBooks(book: List<BookOnDisk>) {
-    dataSource.saveBooks(book)
-      .subscribe(::loadBooks) { e -> Log.e(TAG, "Unable to save books", e) }
-  }
+  BasePresenter<BaseContract.View<Presenter>>(), Presenter {
```

Comment:
We can rip this class to pieces, it isn't even a presenter anymore because it doesn't use a View so calling attach/detach in the reader is wholly pointless. It is just some weird wrapper around DataSource now